File: Tracking System - Api/Models/Trip.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Tracking_System___Api.Dtos;

namespace Tracking_System___Api.Models
{
    public class Trip
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public string Destination { get; set; }
        [NotMapped]
        public location DestinationInMap { get; set; }
        public string Status { get; set; }
        public string CustomerName { get; set; }
        public string CustomerPhone { get; set; }
        public string CustomerAdress { get; set; }
        public string CustomerRegion { get; set; }
        public int DepartmentId { get; set; }
        [ForeignKey("DepartmentId")]
        public virtual Department Department { get; set; }
        public int UserId { get; set; }
        public virtual User user { get; set; }
        [NotMapped]
        public location Start { get; set; }
        public IList<location> current { get; set; }
        [NotMapped]
        public location End { get; set; }

    }
}

File: Tracking-System-Angular/src/app/Models/Trip.ts
import { Customer } from './Customer';
import { Driver } from './Driver';

export class Trip {
id: number;
date: Date;
destination: string;
status: string;
departmentId: number;
customerRegion: string;
customerPhone: string;
customerAdress: string;
customerName: string;


/*
customer: Customer;
driver: Driver;
*/
}
export enum Tripstatus{
    'W' = 'Waiting',
    'D' = 'Done'
 }
